src/components/Input/index.tsx

```diff
@@ -97,30 +157,84 @@ export const Input = React.forwardRef<HTMLInputElement, InputProps>(
       block,
       defaultValue,
       id,
+      onClickCopy = null,
+      shouldShowHideShowButton,
       ...props
     },
     ref
   ) => {
+    const [type, setType] = useState(props.type || 'text')
+    const [wasClicked, setWasClicked] = useState(false)
     const testId = props['data-testid']
     const shouldShowErrorState =
       !(error === false || error === '' || error === undefined) || isUnavailable
     const shouldDisplayInfo = !shouldShowErrorState && info
     const errorMessage = typeof error === 'string' ? error : null
 
+    useEffect(() => {
+      if (wasClicked) {
+        setTimeout(() => {
+          setWasClicked(false)
+        }, 800)
+      }
+    }, [wasClicked])
+
     return (
       <InputContainer $block={block}>
         <Label htmlFor={id} $small={hasSmallLabel}>
           {label}
-          <AsertoInput
-            ref={ref}
-            isValid={isValid && !shouldShowErrorState}
-            isInvalid={shouldShowErrorState}
-            $isUnavailable={isUnavailable}
-            onChange={onChange}
-            style={style}
-            defaultValue={defaultValue}
-            {...props}
-          />
+          <InputButtonBlock>
+            <AsertoInput
+              ref={ref}
+              isValid={isValid && !shouldShowErrorState}
+              isInvalid={shouldShowErrorState}
+              $isUnavailable={isUnavailable}
+              onChange={onChange}
+              style={style}
+              defaultValue={defaultValue}
+              {...props}
+              type={type}
+            />
+            {!(error || isUnavailable || isValid) && (
+              <ButtonsContainer>
+                {shouldShowHideShowButton && (
+                  <>
+                    {type === 'password' ? (
+                      <HideShowButton
+                        variant="secondary-borderless"
+                        onClick={() => setType('text')}
+                        {...mapTestIdToProps(`${testId}-show-btn`)}
+                      >
+                        <img alt="show" src={show} />
+                      </HideShowButton>
+                    ) : (
+                      <HideShowButton
+                        onClick={() => setType('password')}
+                        variant="secondary-borderless"
+                        {...mapTestIdToProps(`${testId}-hide-btn`)}
+                      >
+                        <img alt="hide" src={hide} />
+                      </HideShowButton>
+                    )}
+                  </>
+                )}
+                {onClickCopy && (
+                  <CopyButton
+                    $wasClicked={wasClicked}
+                    variant="secondary-borderless"
+                    onClick={(e) => {
+                      setWasClicked(true)
+                      e.currentTarget.blur()
+                      onClickCopy(String(props.value))
+                    }}
```

Comment:
let's give variables meaningful names
```suggestion
                    onClick={(event) => {
                      setWasClicked(true)
                      event.currentTarget.blur()
                      onClickCopy(String(props.value))
                    }}
```